modules/querier/querier.go

```diff
@@ -626,21 +638,25 @@ func (q *Querier) SearchTagValues(ctx context.Context, req *tempopb.SearchTagVal
 		distinctValues.Collect(v)
 	}
 
-	lookupResults, err := q.forIngesterRings(ctx, userID, nil, func(ctx context.Context, client tempopb.QuerierClient) (interface{}, error) {
-		return client.SearchTagValues(ctx, req)
-	})
-	if err != nil {
-		return nil, fmt.Errorf("error querying ingesters in Querier.SearchTagValues: %w", err)
-	}
-
-outerLoop:
-	for _, resp := range lookupResults {
-		for _, res := range resp.response.(*tempopb.SearchTagValuesResponse).TagValues {
+	forEach := func(ctx context.Context, client tempopb.QuerierClient) error {
+		resp, err := client.SearchTagValues(ctx, req)
+		if err != nil {
+			return err
+		}
+	outerLoop:
+		for _, res := range resp.TagValues {
 			distinctValues.Collect(res)
 			if distinctValues.Exceeded() {
 				break outerLoop // break out of all loops
```

Comment:
I know this wasn't changed in this PR, but it could be a simplified to `return nil` to exit the callback, instead of labeling and breaking the outer loop.

Reply:
removed.